alv-mail-sending-starter/src/main/java/ch/admin/seco/alv/shared/mail/MailIntegrationFlowConfig.java

```
@@ -2,45 +2,79 @@
-		return null;
-	}
+    static final String ERROR_MESSAGE_RECIPIENT_ADDRESS_REJECTED = "450 4.1.2";
+
+    static final String[] ERROR_MESSAGES_TO_IGNORE = new String[] {
+            ERROR_MESSAGE_RECIPIENT_ADDRESS_REJECTED
+    };
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MailIntegrationFlowConfig.class);
+
+    private final JavaMailSender mailSender;
+
+    private final QueueChannel inputChannel;
+
+    private final IntegrationBasisConfig integrationBasisConfig;
+
+    MailIntegrationFlowConfig(JavaMailSender mailSender,
+            MailIntegrationChannels mailIntegrationChannels,
+            IntegrationBasisConfig integrationBasisConfig) {
+        this.mailSender = mailSender;
+        this.inputChannel = mailIntegrationChannels.mailInputChannel();
+        this.integrationBasisConfig = integrationBasisConfig;
+    }
+
+    @Bean
+    IntegrationFlow mailSendingFlow() {
+        return IntegrationFlows
+                .from(this.inputChannel)
+                .transform(MailPreparator::new,
+                        c -> c
+                                .role(IntegrationBasisConfig.DEFAULT_INTEGRATION_ROLE_NAME)
+                                .poller(this.integrationBasisConfig.pollerFactory().builder().retryAware(true).build())
+                )
+                .handle(this::sendMail)
+                .get();
+    }
+
+    private Object sendMail(MailPreparator mailPreparator, Map<String, Object> headers) {
+        try {
+            mailSender.send(mailPreparator);
+        } catch (MailException e) {
+            handleMailException(e);
+        }
+        return null;
+    }
+
+    private void handleMailException(MailException e) {
+        String message = NestedExceptionUtils.getMostSpecificCause(e).getMessage();
+
+        for (String messageToIgnore : ERROR_MESSAGES_TO_IGNORE) {
+            if (message.contains(messageToIgnore)) {
+                LOGGER.info("Skipping email with exception message: {}", message);
```

@pavax Can we throw away the invalid emails or should we route them to the dead letter queue?

---

we we'd have to investigate much deeper into how we can saftely re-route that mail to let's say a differnt mail address or to the dlq